returnn/tf/layers/basic.py

```diff
@@ -4292,18 +4292,24 @@ def get_out_data_from_opts(cls, name, sources, red1=-1, red2=-2, var1=-2, var2=-
     a_rem_dims = [a_shape[i] for i in a_rem_axes]
     a_var_dims = [a_shape[i] for i in a_var_axes]
     b_var_dims = [b_shape[i] for i in b_var_axes]
-    time_dim_axis = None
-    if a_out.time_dim_axis is not None:
-      time_dim_axis = cls._axis1_to_output(a_out.time_dim_axis, a_rem_axes=a_rem_axes, a_var_axes=a_var_axes)
-    if time_dim_axis is None and b_out.time_dim_axis is not None:
-      time_dim_axis = cls._axis2_to_output(
-        b_out.time_dim_axis, b_rem_axes=b_rem_axes, a_var_axes=a_var_axes, b_var_axes=b_var_axes)
-    if time_dim_axis is None and (a_out.time_dim_axis is not None or b_out.time_dim_axis is not None):
-      # We had some time dim axis before and reduced it now.
-      # But maybe there are others, so let's automatically figure out.
-      time_dim_axis = NotSpecified
+
+    def find_axis(a_axis, b_axis):
+      if a_axis is not None:
+        return cls._axis1_to_output(a_axis, a_rem_axes=a_rem_axes, a_var_axes=a_var_axes)
```

Comment:
The logic slightly changed here. It should not directly return, right? It should check if the output from `_axis1_to_output` is not None, and only return in that case. Otherwise, it should go on with the further code.